### PR TITLE
Code Improvements: added a few minor fixes, logger code improved, added unittests

### PR DESCRIPTION
**Pull request by vivekkumac**
**Code Fixes:**
Moved a few host basic cli calls to ACK plugin
Handling ACK Plugin output parsing more robustly

**Logger Code:**
Now logger is intiated in ack_cli with log rotation. Functionality
 remains same with more stability except logger now logs ACK and XS
 versions in beginning. Also fixes a few poorly coded segments.

**Unittests:**
test_get_xenserver_version
test_logging_methods
test_get_xenserver_version
test_get_xcp_version
test_get_system_info

**Others:**
Added unittest config.py to store real data about setup
Added bigger outputs and expected result in files under rawdata dir in unittest dir.
Also updated XenAPI Mock correspondingly.

### CODE REVIEW

File: acktools/tests/acktools_tests.py [hunk 1]
```diff
@@ -8,17 +8,14 @@
...
 
-    returncode = 0
-    stderr = None
-
     def __init__(self, output, err=None):
         self.output = output
         if err:
-            self.returncode = 1
-            self.stderr = err
+            self.returncode = 1 if err else 0
```
Comment (bpwook) [inline acktools/tests/acktools_tests.py]: This is in 'if err' block. Probably use 'else' block properly.

File: acktools/tests/acktools_tests.py [hunk 2]
```diff
@@ -8,17 +8,14 @@
...
+            self.__stderr = err
```
Comment (bpwook) [inline acktools/tests/acktools_tests.py]: This is in 'if err' block. Probably you need initialising.

File: acktools/tests/acktools_tests.py [hunk 3]
```diff
@@ -8,17 +8,14 @@
...
 
     def stderr(self):
```
Comment (bpwook) [inline acktools/tests/acktools_tests.py]: If you prefer you can make this read only property.

**Comment by vivekkumac** [comment]
@bpwook I have updated code with review changes. Please review again.

File: autocertkit/tests/xenapi_mock.py [hunk 4]
```diff
@@ -489,8 +499,25 @@ def __init__(self, host):
     def __getattr__(self, name):
         return self.__defaultOutput
 
-    def __defaultOutput(self, *args):
+    def __defaultOutput(self, args):
```
Comment (vivekkumac) [inline autocertkit/tests/xenapi_mock.py]: This change was done to match ack plugin calls. args is actually a dictionary and is passed in plugin as it is. We always pass args, either as {} or {key1:val1 , key2:val2 ....}

**Pull request merged**